src/app/stats/stats.component.ts

```diff
@@ -7,9 +34,78 @@ import { Component, OnInit } from '@angular/core';
 })
 export class StatsComponent implements OnInit {
 
-  constructor() { }
+  inProgress = false;
+  nodes: Array<GraphVertex> = [];
+  links: Array<GraphEdge> = [];
 
-  ngOnInit() {
+  constructor(
+    private readonly blockService: BlockTreeService,
+    private readonly progress: ProgressService,
+  ) {
+    this.progress.progress.subscribe(v => this.inProgress = v);
   }
 
+  // slotPoll() {
+  //   return interval(SLOT_DURATION_MILLISECONDS)
+  //     .pipe(
+  //       switchMap(() => {
+  //         this.progress.startProgress();
+  //         return this.blockService.getBlockTree();
+  //       }), 
+  //       // distinctUntilChanged(), // only emit and redraw graph if the value has actually changed
+  //     );
+  // }
+
+  ngOnInit() {
+    this.blockService.getBlockTree().subscribe(async (res: BlockTreeResponse) => {
+      const tree = this.blockService.sortTree(res.getTreeList());
+      const existingParentBlocks = {};
+      this.nodes = tree.map((node, idx) => {
+        const blockRoot = this.blockService.toHexString(node.getBlockRoot());
+        existingParentBlocks[blockRoot] = true;
+
+        let participatedVotesString = Math.floor(node.getParticipatedVotes() / MAX_DEPOSIT_AMOUNT).toString();
+        if (participatedVotesString.indexOf('.') === -1) {
+          participatedVotesString += '.';
+        }
+        while (participatedVotesString.length < participatedVotesString.indexOf('.') + 15) {
+          participatedVotesString += '0';
+        }
+        let totalVotesString = Math.floor(node.getParticipatedVotes() / MAX_DEPOSIT_AMOUNT).toString();
+        if (totalVotesString.indexOf('.') === -1) {
+          totalVotesString += '.';
+        }
+        while (totalVotesString.length < totalVotesString.indexOf('.') + 15) {
+          totalVotesString += '0';
+        }
+        return {
+          id: blockRoot,
+          label: blockRoot,
+          isJustified: idx === 0,
+          participatedVotes: participatedVotesString,
+          totalVotes: totalVotesString,
+          numAttestations: node.getBlock().getBody().getAttestationsList().length,
+          numDeposits: node.getBlock().getBody().getDepositsList().length,
+          slot: (new BigNumber(node.getBlock().getSlot()).minus(new BigNumber(GENESIS_SLOT))).toString(),
+        };
+      });
+
+      for (let i = 0; i < this.nodes.length; i++) {
+        const node = tree[i];
```

Comment:
```suggestion
        const node = this.nodes[i];
```
?

Also, any reason why not using
`for (const item of items)`

Reply:
+1 to `for (const node of this.nodes)`